apps/picsa-tools/monitoring-tool/src/app/components/access-code-dialog/access-code-dialog.component.ts

```diff
@@ -0,0 +1,102 @@
+import { CommonModule } from '@angular/common';
+import { Component, Inject, OnDestroy, signal } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MAT_DIALOG_DATA, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { PicsaTranslateModule } from '@picsa/shared/modules';
+
+export interface AccessCodeDialogData {
+  formTitle: string;
+  accessCode: string;
+}
+
+export interface AccessCodeDialogResult {
+  success: boolean;
+  code?: string;
+}
+
+@Component({
+  selector: 'monitoring-access-code-dialog',
+  templateUrl: './access-code-dialog.component.html',
+  styleUrls: ['./access-code-dialog.component.scss'],
+  standalone: true,
+  imports: [
+    CommonModule,
+    FormsModule,
+    MatButtonModule,
+    MatDialogModule,
+    MatFormFieldModule,
+    MatIconModule,
+    MatInputModule,
+    MatSnackBarModule,
+    PicsaTranslateModule,
+  ],
+})
+export class AccessCodeDialogComponent implements OnDestroy {
+  accessCode = '';
+  hidePassword = signal(true);
+  showError = signal(false);
+
+  private errorTimer: any;
```

Comment:
nit(blocking): definitely nice to have the error remove, although I think yo ucan probably just clear it when the user starts typing into the input. E.g. binding to the `(change)` event on the input and just use to clear errors if they exist. That way you won't need to worry about the OnDestroy callback either